npsat_manager/views.py

```diff
@@ -112,3 +177,27 @@ class ModificationViewSet(viewsets.ModelViewSet):
 
 	def get_queryset(self):
 		return models.Modification.objects.filter(model_run__user=self.request.user).order_by('id')
+
+
+class ResultPercentileViewSet(viewsets.ModelViewSet):
+	"""
+	API endpoint for model results
+	restricted to only allow GET request
+
+	Permission: same as the model run, must be authenticated
+	"""
+	permission_classes = [IsAuthenticated]
+	http_method_names = ["get"]
+
+	serializer_class = serializers.ResultPercentileSerializer
+
+	def get_queryset(self):
+		return models.ResultPercentile.objects\
+			.select_related('model')\
+			.filter(
+				Q(model__user=self.request.user) |
```

Comment:
not sure we need to use Q objects in here, or that we need the select_related. Should just be able to do the direct filtering. 

Also, looks like we're returning all results every time any new results are requested? I think it'd be good if we had it structured so that there was an endpoint for all objects, but also to get a single set of results (so that a user with 20 model runs can run a new model, and we only need to retrieve the new results from the client?) - not a big deal, but an enhancement to consider.

Reply:
This will first filter out all results available to a user. The model result doesn't have any field indicating whom it belongs to, so by including `model`, we can find the model creator who owns the results. This queryset basically filters results that are owned by the user, are public, or are base model results.

Currently, this endpoint is only used when querying single model results, but I can make it support multiple models results query.

Reply:
Totally, we definitely need to traverse the relationship and I made a mistake in saying we don't need the Q objects - I forgot that we'd need them for an OR query like this. That said, you shouldn't need the `select_related` call - that might trigger an extra DB query we don't need. It should work fine without it, but let me know if it doesn't.

Reply:
Oh, didn't realize that we can use `model` directly without `select_related`. I must have messed with ORM and raw SQL. Thanks!